### ISSUE TITLE
Project Wizard Smoke Tests

### ISSUE THREAD

**Issue by sharon-wang** [posit-dev/positron#3879]
### R project
- [x] R Project with Renv Environment https://github.com/posit-dev/positron/pull/3881 

### Python project
- [x] Python Project using existing python interpreter with ipykernel already installed https://github.com/posit-dev/positron/pull/3951
- [x] Python Project using existing python interpreter with ipykernel **not** installed https://github.com/posit-dev/positron/pull/3951
- [x] Python Project with new Conda environment https://github.com/posit-dev/positron/pull/4042
- ~Python Project with new Venv environment~ this is already covered in the basic Python test

### Misc
- [x] New project with git init selected https://github.com/posit-dev/positron/pull/4044
- [x] Document installing Conda in smoke test README https://github.com/posit-dev/positron/pull/4066
- [x] Timing fixes for python env provider and interpreter dropdowns when Conda is selected quickly https://github.com/posit-dev/positron/pull/4066

**Comment by sharon-wang** [comment]
I'll be merging https://github.com/posit-dev/positron/pull/3951, https://github.com/posit-dev/positron/pull/4042 and https://github.com/posit-dev/positron/pull/4044 after addressing merge conflicts.

There are some enhancements to handling timing in the smoke tests and a separate fix for the Python interpreter dropdown in the Project Wizard that I'll add separately.

**Comment by sharon-wang** [comment]
This is now complete. Tests passed on `main` after these changes were merged. https://github.com/posit-dev/positron/actions/runs/10002371271/job/27647541691

**Comment by sharon-wang** [comment]
Some extra notes around the **Jupyter Notebook** project wizard default smoke test:

- Both Jupyter and R default tests were often timing out in CI (Linux with limited resources) when waiting for the console to be ready after initializing a new project. Eventually, I had to remove the `waitForConsole` check to make progress on the smoke tests. See https://github.com/posit-dev/positron/pull/3951/commits/b3b0ae8a971537787ea08344c29c4bced909ca2c -- adding the `waitForConsole`s back in should reproduce the error in CI.
- Later, I wanted to add the `git init` project wizard test to the Jupyter Notebook suite, but it was still timing out after the default test, so I ended up adding the test to Python instead, see https://github.com/posit-dev/positron/pull/4044
- This timeout issue may be related to https://github.com/posit-dev/positron/issues/3779?
- The tests were timing out due to hanging on `Starting up...`, although the rest of the project initialization tasks seemed to succeed
